apps/pre-award/copilot/environments/addons/communities-redirect-cloudfront-function.yml

```diff
@@ -0,0 +1,72 @@
+Parameters:
+  App:
+    Type: String
+    Description: Your application's name.
+  Env:
+    Type: String
+    Description: The environment name your service, job, or workflow is being deployed to.
+
+Mappings:
+  DomainSuffixMap:
+    dev:
+      "DomainSuffix": ".dev.access-funding.communities.gov.uk"
+    test:
+      "DomainSuffix": ".test.access-funding.communities.gov.uk"
+    uat:
+      "DomainSuffix": ".uat.access-funding.communities.gov.uk."
+    prod:
+      "DomainSuffix": ".access-funding.communities.gov.uk"
+  
+Resources:
+  CommunitiesRedirect:
+    Type: 'AWS::CloudFront::Function'
+    Properties:
+      Name: CommunitiesRedirect
+      AutoPublish: true
+      FunctionCode: !Sub
+      - |
+        function handler(event) {
+          const host = event.request.headers.host.value;
+          const uri = event.request.uri;
+          
+          var newUrl;
+          // renamed services
+          if (host.startsWith('frontend.') && host.endsWith('.levellingup.gov.uk'))
+              newUrl = `https://apply${DomainSuffix}${!uri}`;
+          else if (host.startsWith('authenticator.') && host.endsWith('.levellingup.gov.uk'))
+              newUrl = `https://account${DomainSuffix}${!uri}`;
+          else if (host.startsWith('assessment.') && host.endsWith('.levellingup.gov.uk'))
+            newUrl = `https://assess${DomainSuffix}${!uri}`;
+          else if (host.startsWith('forms.') && host.endsWith('levellingup.gov.uk'))
+            newUrl = `https://application-questions${DomainSuffix}${!uri}`;
+          // unchanged named services
+          else if (host.endsWith('.levellingup.gov.uk')) {
+            const serviceHostName = host.split('.')[0]
+            newUrl = `${!serviceHostName}${DomainSuffix}${!uri}`
+          }
+          else {
+          // non-legacy domain pass through unchanged request
+              return event.request
+          }
```

Comment:
Is totally fine, but might read easier to just do `host.endsWith('communities.gov.uk')` once at the start rather than check for levellingup every time?